ros2bag/ros2bag/api/__init__.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from argparse import ArgumentTypeError
+import os
+from typing import Dict
+from typing import Optional
+
+from rclpy.duration import Duration
+from rclpy.qos import QoSDurabilityPolicy
+from rclpy.qos import QoSHistoryPolicy
+from rclpy.qos import QoSLivelinessPolicy
+from rclpy.qos import QoSProfile
+from rclpy.qos import QoSReliabilityPolicy
+
+# This map needs to be updated when new policies are introduced
+_QOS_POLICY_FROM_SHORT_NAME = {
+    'history': QoSHistoryPolicy.get_from_short_key,
+    'reliability': QoSReliabilityPolicy.get_from_short_key,
+    'durability': QoSDurabilityPolicy.get_from_short_key,
+    'liveliness': QoSLivelinessPolicy.get_from_short_key
+}
+_DURATION_KEYS = ['deadline', 'lifespan', 'liveliness_lease_duration']
+_ENUM_KEYS = ['history', 'reliability', 'durability', 'liveliness']
+_VALUE_KEYS = ['depth', 'avoid_ros_namespace_conventions']
+
+
+def is_dict_valid_duration(duration_dict: Dict[str, int]) -> bool:
+    """Check if dictionary contains the keys `sec` and `nsec`."""
+    return all(key in duration_dict for key in ['sec', 'nsec'])
+
+
+def dict_to_duration(time_dict: Optional[Dict[str, int]]) -> Duration:
+    """Convert a QoS duration profile from YAML into an rclpy Duration."""
+    if time_dict:
+        try:
+            return Duration(seconds=time_dict.get('sec'), nanoseconds=time_dict.get('nsec'))
+        except TypeError:
+            raise ValueError(
+                'Time overrides must include both seconds (sec) and nanoseconds (nsec).')
+    else:
+        return Duration()
```

Comment:
As-is this won't throw because you use `get`, and you're catching the wrong type, it would be a `KeyError`.

```suggestion
def dict_to_duration(time_dict: Optional[Dict[str, int]]) -> Duration:
    try:
        return Duration(seconds=time_dict['sec'], nanoseconds=time_dict['nsec']) if time_dict else Duration()
    except KeyError:
        raise ValueError('Durations require both keys "sec" and "nsec")
```

Let's add some unit tests for these utilities

Reply:
Duration throws a `TypeError` if the nano/seconds is none. Your approach is also valid tho.